core/src/main/java/io/grpc/internal/GrpcAttributes.java

```diff
@@ -50,6 +50,13 @@
   public static final Attributes.Key<Boolean> ATTR_LB_PROVIDED_BACKEND =
       Attributes.Key.create("io.grpc.grpclb.lbProvidedBackend");
 
+  /**
+   * The authority to be used when constructing {@link io.grpc.Channel}s to this this EquivalentAddressGroup.
+   */
+  @EquivalentAddressGroup.Attr
+  public static final Attributes.Key<String> ATTR_CHANNEL_AUTHORITY =
```

Comment:
This should be a public API, not in internal. @zhangkun83, is the only place we have for things like this `io.grpc.Grpc`?

Reply:
Yes, the last time I checked `io.grpc.Grpc` is the only place where these Attrs are defined (and they all are prefixed `TRANSPORT_ATTR_`) . I guess this can also be considered a `TRANSPORT_ATTR_` ?

Reply:
I would suggest we name it as `ATTR_AUTHORITY_OVERRIDE` and put it in `EquivalentAddressGroup`. `TRANSPORT_ATTR_` is for pulling information out of transport, while this attribute is passing information to the transport, which is different. It's better to put the attributes close to where it's used. `TRANSPORT_ATTR_` is in `Grpc` because there is no other public place where we can put them, since transport API is internal.

Reply:
Makes sense. Agreed on all points.

Reply:
Renamed the attribute as suggested by @zhangkun83 . I am assuming that 

>  put it in `EquivalentAddressGroup`

means using the `@EquivalentAddressGroup.Attr` annotation as I already did?

Reply:
No, I think he meant move the field to `io.grpc.EquivalentAddressGroup`, directly.

Reply:
ok. done.